lit-base/model_service.py

```diff
@@ -28,7 +28,11 @@ def preprocess(event):
             # download image from url
             with tempfile.NamedTemporaryFile(delete=True) as img_file:
                 if url:
-                    req = urllib2.urlopen(url)
+                    try:
+                        req = urllib2.urlopen(url)
+                    except urllib2.URLError:
+                        time.sleep(0.01)
+                        req = urllib2.urlopen(url)
```

Comment:
Thanks for adding the retry logic here. I think instead of looping using the ```except``` clause it might be more intuitive to use a while loop. That way you can also make the number of retries more flexible.

Reply:
Also can you make the retry counter and retry interval configurable with defaults. So the User can control whether he wants a retry, how long or just set the counter to 1 and you don't retry.

For the case of Lambda, I think retry adds additional latency and the user wouldn't know if it is because a Model file is throttled by S3 or model unavailable intermittently due to the hosting, but we can let the user decide using configuration.

Reply:
pseudo-code for exponential backoff. 
```
retry = 0
while (retry++ < retrycount):
	get_url
if Not available:
	sleep( baseMilliseconds * 2^retry)
```